lib/kube/proxy/forwarder.go

```diff
@@ -389,7 +395,36 @@ func (f *Forwarder) withAuth(handler handlerWithAuthFunc) httprouter.Handle {
 			return nil, trace.Wrap(err)
 		}
 		return handler(authContext, w, req, p)
-	})
+	}, f.formatResponseError)
+}
+
+func (f *Forwarder) formatForwardResponseError(rw http.ResponseWriter, r *http.Request, respErr error) {
+	f.formatResponseError(rw, respErr)
+}
+
+func (f *Forwarder) formatResponseError(rw http.ResponseWriter, respErr error) {
+	status := &metav1.Status{
+		Status: metav1.StatusFailure,
+		// Don't trace.Unwrap the error, in case it was wrapped with a
+		// user-friendly message. The underlying root error is likely too
+		// low-level to be useful.
+		Message: respErr.Error(),
+		Code:    int32(trace.ErrorToCode(respErr)),
+	}
+	data, err := runtime.Encode(statusCodecs.LegacyCodec(), status)
+	if err != nil {
+		f.log.Warningf("Failed encoding error into kube Status object: %v", err)
+		trace.WriteError(rw, respErr)
+		return
+	}
+	rw.Header().Set("Content-Type", "application/json")
+	// Always write InternalServerError, that's the only code that kubectl will
```

Comment:
nit: Can you link to the code where kubectl does this so we might have an anchor for future updates.

Reply:
`kubectl` code was inscrutable to me, due to a thick layer of abstractions and libraries. I don't know where this is handled.
I figured out this behavior after several hours of reverse-engineering the behavior.